Mage.Sets/src/mage/cards/n/Nalfeshnee.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.n;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfEndStepTriggeredAbility;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeSourceEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterSpell;
+import mage.filter.predicate.card.CastFromZonePredicate;
+import mage.filter.predicate.mageobject.MageObjectReferencePredicate;
+import mage.game.Game;
+import mage.game.stack.Spell;
+import mage.game.stack.StackObject;
+import mage.players.Player;
+import mage.util.functions.StackObjectCopyApplier;
+
+import java.util.UUID;
+
+/**
+ * @author Grath
+ */
+public final class Nalfeshnee extends CardImpl {
+
+    private static final FilterSpell filter = new FilterSpell("a spell from exile");
+
+    static {
+        filter.add(new CastFromZonePredicate(Zone.EXILED));
+    }
+
+    public Nalfeshnee(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{5}{R}");
+
+        this.subtype.add(SubType.BEAST);
+        this.subtype.add(SubType.DEMON);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(6);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Whenever you cast a spell from exile, copy it. You may choose new targets for the copy. If it’s a permanent spell, the copy gains haste and “At the beginning of the end step, sacrifice this permanent.”
+        this.addAbility(new SpellCastControllerTriggeredAbility(new NalfeshneeEffect(), filter, false));
+    }
+
+    private Nalfeshnee(final Nalfeshnee card) {
+        super(card);
+    }
+
+    @Override
+    public Nalfeshnee copy() {
+        return new Nalfeshnee(this);
+    }
+}
+
+class NalfeshneeEffect extends OneShotEffect {
+
+    NalfeshneeEffect() {
+        super(Outcome.Benefit);
+        staticText = "copy it, You may choose new targets for the copy. If it's a permanent spell, the copy gains haste and \"At the beginning of the end step, sacrifice this permanent.\"";
```

Comment:
The first part of this effect's text should be "copy it. You may choose..", as opposed to "copy it, You may choose..", note the period instead of the comma

Reply:
Done.